retriever/try_install_modified.py

```diff
@@ -0,0 +1,139 @@
+"""Attempt to update datasets in all database management systems.
+This module, when run, attempts to install datasets from modified Retriever
+scripts in the /scripts folder (except for those listed in IGNORE), for each
+engine in ENGINE_LIST() from __init__.py. In other words, it trys to install
+using all possible combinations of database platform and script and checks
+to see if there are any errors. It does not check the values in the database.
+"""
+from __future__ import print_function
+from __future__ import absolute_import
+import os
+import sys
+from imp import reload
+from retriever.lib.tools import choose_engine
+from retriever import MODULE_LIST, ENGINE_LIST, SCRIPT_LIST
+from retriever.version import get_module_version, update_version_file
+
+
+def get_modified_scripts():
+    """Get modified script list compared to version.txt."""
+    modified_scripts = []
+    script = get_module_version()
+    with open("version.txt", "r") as version_file:
+        version_file.readline()
+        for counter, line in enumerate(version_file.readlines()):
```

Comment:
Test out this version of def get_modified_scripts():

```
def to_string(value):
    if sys.version_info >= (3, 0, 0):
        return value.decode("UTF-8")
    else:
        return value

def get_modified_scripts():
    subprocess.check_output(
    'git fetch https://github.com/weecology/retriever.git master 2>/dev/null',
    shell=True)

    output = subprocess.check_output(
                "git diff --name-only FETCH_HEAD ../scripts",
                shell=True)

    # get list of changed scripts
    scripts_changed = to_string(output).splitlines()

    return scripts_changed
```

This will grab all modified scripts relative to upstream/master, like we do in hooks/precommit, and relieves us from having to call get_module_version() with slightly more readable code. Having a method like get_modified_scripts() could be very useful though in the future, but we should run it through @henrykironde first before we make changes to the retriever __init__ file.